test/integration/puppetlabs/services/jruby/jruby_metrics_service_test.clj

```diff
@@ -886,7 +886,7 @@
               (is (= 1 (.getCount timer))
                   (str "Timer has a accumulated a count for borrow reason: " timer-name)))))))))
 
-(deftest ^:metrics borrow-timeout-test
+(deftest ^:metrics ^:single-threaded-only borrow-timeout-test
```

Comment:
I'm curious how you decided which tests should be single-threaded only. 

This test in particular is one I was using to test my worker-id stuff, so seems like it could work in multithreaded?

Reply:
I flagged every test that currently fails in `multithreaded` mode. The metrics tests in particular I assumed were going to be shortly fixed by your work. You should feel free to delete those tags once the tests pass in both modes.

Some of the others need alternative versions written (e.g. the ones that test for different variables being set in different JRuby instances), but I wanted to do that as a separate ticket to keep this manageable to review.

Reply:
gotcha! cool, so then as long as there are tickets with the necessary follow-up work, this makes sense to me.

Reply:
Is there (tickets?)? Or is it just that whichever of you two has their PR merged second should change this to not have a thread label?

Reply:
I'll remove this one on my PR, since I think this PR should go first.
But we should ticket the rest.